Source/Server/Player.c

```diff
@@ -149,22 +149,22 @@ void set_player_respawn_point(server_t* server, player_t* player)
         player->movement.forward_orientation.z = 0.f;
 
         // Find nearest spawn point
-        for(uint16_t z = player->movement.position.z; z < server->s_map.map.size_z; z++) {
-            if(is_valid_spawn_point(server, player->movement.position.x, player->movement.position.y, z)) {
-                player->movement.position.z = z;
+        for (uint16_t z = player->movement.position.z; z < server->s_map.map.size_z; z++) {
+            if (is_valid_spawn_point(server, player->movement.position.x, player->movement.position.y, z)) {
+                player->movement.position.z = z - 2.4F;
```

Comment:
Ref: https://github.com/piqueserver/piqueserver/blob/b14f6f00ca021ef724cb72e975bd8228a25dcc89/pyspades/player.py#L847C20-L847C20

I'm not sure about this up and down looping?